packages/next/src/client/dev/dev-build-watcher.ts

```diff
@@ -1,11 +1,20 @@
 /* eslint-disable @typescript-eslint/no-use-before-define */
+import {
+  HMR_ACTIONS_SENT_TO_BROWSER,
+  HMR_ACTION_TYPES,
+} from '../../server/dev/hot-reloader-types'
 import { addMessageListener } from './error-overlay/websocket'
 
 type VerticalPosition = 'top' | 'bottom'
 type HorizonalPosition = 'left' | 'right'
 
+export interface ShowHideHandler {
+  show: () => void
+  hide: () => void
+}
+
 export default function initializeBuildWatcher(
-  toggleCallback: (cb: (obj: Record<string, any>) => void) => void,
+  toggleCallback: (handlers: ShowHideHandler) => void,
```

Comment:
Changed this as previously it relied on `building` and `built` faked HMR events which wasn't needed, just a way to show/hide the building indicator.